waku/wms.md

```diff
@@ -45,13 +45,25 @@ limit  = 4OCTET
 ; array of a cursor returned from the previous request (optional)
 cursor = 32OCTET
 
-payload = "[" lower upper bloom limit [ cursor ] "]"
+; List of topics interested in
+topic-interest   = "[" *1000topic "]"
+
+; 4 bytes of arbitrary data
+topic = 4OCTET
+
+payload_alt1 = "[" lower upper bloom limit [ cursor ] "]"
+
+payload_alt2 = "[" lower upper bloom limit cursor [ topic-interest ] "]"
+
+payload = payload_alt1 | payload_2
 ```
 
 The `Cursor` field SHOULD be filled in if a number of envelopes between `Lower` and `Upper` is greater than `Limit` so that the requester can send another request using the obtained `Cursor` value. What exactly is in the `Cursor` is up to the implementation. The requester SHOULD NOT use a `Cursor` obtained from one mailserver in a request to another mailserver because the format or the result MAY be different.
 
 The envelope MUST be encrypted with a symmetric key agreed between the requester and Mailserver.
 
+If `topic-interest` is used `Cursor` field MUST be specified for the argument order to be unambiguous. However, it MAY be set to null. `topic-interest` is used to specify limit envelopes to. If this is specified, a mailserver MUST NOT send messages that aren't in in that topic. This option takes precedence over the `bloom` parameter.
```

Comment:
I don't understand what you mean by "`topic-interest` is used to specify limit envelopes to."

Reply:
Also, what does `precedence` mean here exactly? I think this can still be wrongly interpreted. Perhaps it should be written a bit stronger, something like: `When topic-interest is set, the bloom filter parameter should be ignored`

Reply:
And to add on the previous. Perhaps a bit of a weird and special case but what if we pass an empty topic-interest array. No messages may be send right? Or bloom applies?

Reply:
Fixed up language, made phrasing stronger and included edge case. PTAL